code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -99,26 +102,27 @@
 	var/list/cached_gases = air.gases //this speeds things up because accessing datum vars is slow
 	var/temperature = air.temperature
 	var/burned_fuel = 0
-
+	var/list/cached_results = air.reaction_results
```

Comment:
Shouldn't cache the list if the reaction doesn't run

Reply:
oh right woops